test/ruby/test_set.rb

```diff
@@ -864,6 +864,19 @@ def test_reset
       assert_equal(klass.new([a]), set, klass.name)
     }
   end
+
+  def test_set_gc_compact_does_not_allocate
+    assert_in_out_err([], <<-"end;", [], [])
+    def x
+      s = Set.new
+      s << Object.new
+      s
+    end
+
+    m = x
+    Process.warmup
```

Comment:
```suggestion
    GC.compact
```

nitpick: Could just be `GC.compact`.